templates/terraform/iam_policy.go.erb

```diff
@@ -260,8 +260,9 @@ func (u *<%= resource_name -%>IamUpdater) qualify<%= object.name -%>Url(methodId
 	return fmt.Sprintf("<%= object.__product.base_url -%>%s<%= object.iam_policy.method_name_separator -%>%s", fmt.Sprintf("<%= import_url -%>", <%= string_qualifiers -%>), methodIdentifier)
 }
 
+<% id_format = object.id_format || object.self_link_uri -%>
```

Comment:
There isn't a great place to stick this logic, but wdyt about throwing this in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform.rb instead of at each place we use it? That way if we change it in the future, we don't need to change many places (and we can stick a comment on it about the expected shape of ids for people reading the code)